finality-aleph/src/metrics/transaction_pool.rs

```diff
@@ -0,0 +1,247 @@
+
+    pub struct TestTransactionPoolSetup {
+        pub client: Arc<TestClient>,
+        pub pool: Arc<FullTransactionPool>,
+        pub proposer_factory: TProposerFactory,
+        pub transaction_pool_info_provider: TransactionPoolWrapper<TChainApi, TBlock>,
+    }
+
+    impl TestTransactionPoolSetup {
+        pub fn new(client: Arc<TestClient>) -> Self {
+            let spawner = sp_core::testing::TaskExecutor::new();
+            let pool = BasicPool::new_full(
+                Default::default(),
+                true.into(),
+                None,
+                spawner.clone(),
+                client.clone(),
+            );
+            let transaction_pool_info_provider = TransactionPoolWrapper::new(pool.clone());
+
+            let proposer_factory =
+                ProposerFactory::new(spawner, client.clone(), pool.clone(), None, None);
+
+            TestTransactionPoolSetup {
+                client,
+                pool,
+                proposer_factory,
+                transaction_pool_info_provider,
+            }
+        }
+
+        pub async fn propose_block(&mut self, at: THash, weight_limit: Option<usize>) -> TBlock {
+            let proposer = self
+                .proposer_factory
+                .init(&self.client.expect_header(at).unwrap())
+                .await
+                .unwrap();
+
+            let block = proposer
+                .propose(
+                    Default::default(),
+                    Default::default(),
+                    Duration::from_secs(30),
+                    weight_limit,
+                )
+                .await
+                .unwrap()
+                .block;
+
+            self.import_block(block).await
+        }
+
+        pub async fn import_block(&mut self, block: TBlock) -> TBlock {
+            let stream = self.client.every_import_notification_stream();
+            self.client
+                .import(BlockOrigin::Own, block.clone())
+                .await
+                .unwrap();
+
+            let notification = stream
+                .filter(|notification| future::ready(notification.hash == block.hash()))
+                .next()
+                .await
+                .expect("Notification was sent");
+
+            if notification.is_new_best {
+                self.pool.maintain(notification.try_into().unwrap()).await;
+            }
+
+            block
+        }
+
+        pub async fn finalize(&mut self, hash: THash) {
+            let stream = self.client.finality_notification_stream();
+            self.client.finalize_block(hash, None).unwrap();
+            let notification = stream
+                .filter(|notification| future::ready(notification.hash == hash))
+                .next()
+                .await
+                .expect("Notification was sent");
+
+            self.pool.maintain(notification.into()).await;
+        }
+
+        pub fn xt(
```

Comment:
```suggestion
        pub fn extrinsic(
```

Reply:
Didn't know you were fighting with `cargo fmt` eating space behind the scenes :stuck_out_tongue:.